src/common/transformations/src/transformations/common_optimizations/reshape_sequence_fusion.cpp

```diff
@@ -58,8 +58,15 @@ ngraph::pass::ReshapeSequenceFusion::ReshapeSequenceFusion() {
             input = node->input_value(0);
         }
 
-        reshape->input(0).replace_source_output(input);
-        copy_runtime_info(nodes, reshape);
+        // remove redundant reshapes
+        if (input.get_node_shared_ptr()->get_output_partial_shape(0).is_static() && reshape->get_output_partial_shape(0).is_static() &&
+            input.get_node_shared_ptr()->get_output_shape(0) == reshape->get_output_shape(0)) {
+            return replace_output_update_name(reshape->output(0), input);
+        } else {
+            reshape->input(0).replace_source_output(input);
+            copy_runtime_info(nodes, reshape);
+        }
```

Comment:
Input is already an Output<Node> class so you can get partial shape from it.
Also as this transformation is included into MOC pipeline where we can rely on shapes only in case if `use_shape` is true, so we have to parametrize transformation (for example: https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/include/transformations/common_optimizations/split_concat_pair_to_interpolate_fusion.hpp#L32 )

```suggestion
        bool replaced = false;
        if (use_shapes && input.get_partial_shape().is_static() && reshape->get_output_partial_shape(0).is_static() &&
            input.get_shape() == reshape->get_output_shape(0)) {
            // in case if elimination is not allowed we still can eliminate all transposes except last one
            replaced = replace_output_update_name(reshape->output(0), input));
        }
        
        if (!replaced) {
            reshape->input(0).replace_source_output(input);
            copy_runtime_info(nodes, reshape);
        }
```

Also please add tests for this case.

Reply:
Thank you for the explanation!
Done.
About test - Before I added test `ReshapeSequenceFusionEliminate` that covers case when we can remove 2 redundant reorders. I think it's enough